### PR TITLE
[SYCL] Fix failures on tests that use double datatype

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

**Review by steffenlarsen — DISMISSED** [review]

File: SYCL/ESIMD/regression/dgetrf.cpp [hunk 1]
```diff
@@ -6,7 +6,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-// REQUIRES: gpu
+// REQUIRES: gpu, aspect-fp64
```
Comment (AlexeySachkov) [inline SYCL/ESIMD/regression/dgetrf.cpp]: Do we define `aspect-fp64` anywhere? I'm concerned that this change will simply disable those tests

**Review by AlexeySachkov — CHANGES_REQUESTED** [review]

File: SYCL/ESIMD/regression/dgetrf.cpp (hunk 1 above)
Comment (aelovikov-intel) [reply SYCL/ESIMD/regression/dgetrf.cpp]: I've copied from a now obsolete PR, but maybe I didn't fully understand it and missed that fact. Thanks for noticing - I'll verify if it's defined later today.

Comment (aelovikov-intel) [reply SYCL/ESIMD/regression/dgetrf.cpp]: You're totally right - it doesn't exist. I'll change to make runtime `device.has` check.

File: SYCL/SubGroup/info.cpp [hunk 2]
```diff
@@ -24,7 +24,7 @@ int main() {
 
   /* Basic sub-group functionality is supported as part of cl_khr_subgroups
    * extension or as core OpenCL 2.1 feature. */
-  if (!core_sg_supported(Device)) {
+  if (!core_sg_supported(Device) || !Device.has(aspect::fp64)) {
```
Comment (aelovikov-intel) [inline SYCL/SubGroup/info.cpp]:
```suggestion
  if (!core_sg_supported(Device)) {
```

https://github.com/intel/llvm-test-suite/pull/1242 is better than this one.

**Comment by kbobrovs** [comment]
In one of other reviews I mentioned that test reporting "Passed" state w/o actually exercising all features it is supposed to test, can be dangerous - e.g. tests run on TGLLP will report "Passed" w/o running fp64 test cases. Alternative is to extract feature-specific test cases (e.g. fp64-specific) into a separate test which would use "REQUIRES: aspect_fp64" and report "UNSUPPORTED" on TGLLP.

**Comment by aelovikov-intel** [comment]
> Alternative is to extract feature-specific test cases (e.g. fp64-specific) into a separate test which would use "REQUIRES: aspect_fp64" and report "UNSUPPORTED" on TGLLP.

IMO, this is less maintainable and isn't really better. Does anybody even look at UNSUPPORTED on any regular basis? How exactly would that information be used?

**Comment by kbobrovs** [comment]
> > Alternative is to extract feature-specific test cases (e.g. fp64-specific) into a separate test which would use "REQUIRES: aspect_fp64" and report "UNSUPPORTED" on TGLLP.
> 
> IMO, this is less maintainable and isn't really better. Does anybody even look at UNSUPPORTED on any regular basis? How exactly would that information be used?

If nobody looks at this, this is another problem. Taking this to the extreme, a product, which passes all the tests, but all tests return "UNSUPPORTED" is not really tested and should not be released.

**Comment by v-klochkov** [comment]
> > Alternative is to extract feature-specific test cases (e.g. fp64-specific) into a separate test which would use "REQUIRES: aspect_fp64" and report "UNSUPPORTED" on TGLLP.
> 
> IMO, this is less maintainable and isn't really better. Does anybody even look at UNSUPPORTED on any regular basis? How exactly would that information be used?

Even though (static checks + REQUIRES) solution gives more info on which tests run or which skipped, my final vote was for device.has() because IMO using (static checks + REQUIRES) may cause quick growth of number of tests.
2 tests: is what we get after separation of fp64 test cases.
3 tests: if use same approach and separate fp16 test cases to another test.
4 tests: if we have test cases using both fp16&fp64. 
That might grow exponentially with adding more independent/orthogonal features (3 or more).

**Comment by kbobrovs** [comment]
> That might grow exponentially with adding more independent/orthogonal features (3 or more).

I think tests should strive to test feature alone, w/o making combination with other features. So I would not expect exponential growth of number of tests (which would actually be even worse - combinatorial explosion - if we want to test combinations).

**Comment by aelovikov-intel** [comment]
In an offline discussion we've agreed to proceed with `device.has` approach at least short-term.

I've also addressed @AlexeySachkov 's concern in https://github.com/intel/llvm-test-suite/pull/1246/commits/e63b15c1a227339e91e08636d0291f6aabb64ca0. Can somebody please re-review/merge?

**Review by v-klochkov — APPROVED** [review]

**Review by AlexeySachkov — APPROVED** [review]

**Pull request merged**